tracer/src/Datadog.Trace/OTelMetrics/MetricPoint.cs

```diff
@@ -51,6 +71,14 @@ internal void UpdateCounter(double value)
         }
     }
 
+    internal void UpdateObservableCounter(double currentValue)
+    {
+        lock (_histogramLock)
+        {
+            _runningDoubleValue = currentValue;
```

Comment:
nit: For follow-up, we should be able to faithfully report both longs and doubles, so I think we'll need to do some refactoring. For example, a Counter may either be a long or a double, and we should store the intended type